apps/site/src/app/about/getMembers.ts

```diff
@@ -1,28 +1,14 @@
-const FEED_URL = "https://docs.google.com/spreadsheets/d/";
-const SPREADSHEET_KEY = "1DWCOQBlzA3mpa2BXYXPmQrF9_-SpoPFbTdDQuCQ83hU";
-const QUERY = "pub";
-const FORMAT = "output=tsv";
-
-const dataURL = FEED_URL + SPREADSHEET_KEY + "/" + QUERY + "?" + FORMAT;
-
-const getSheetsData = async (page: string) => {
-	const response = await fetch(dataURL);
-
-	if (response.status !== 200)
-		console.warn(
-			"Error ocurred while fetching schedule data:",
-			response.statusText
-		);
-
-	const csv = await response.text();
-	for (const line of csv.split("\n")) {
-		const [key, value] = line.split("\t");
-		if (key === page) {
-			return JSON.parse(value);
-		}
-	}
-
-	return null;
-};
-
-export const getMembers = async () => await getSheetsData("members");
+import { cache } from "react";
+import { client } from "@/lib/sanity/sanityClient";
+
+export const getMembers = cache(async () => {
+	return await client
```

Comment:
The fetched content should be validated with [zod](https://zod.dev/) to ensure that it's the right shape and has a type. You can reference the [getEvents function](https://github.com/HackAtUCI/hack.ics.uci.edu/blob/4802377eb66c76732584d7708f356f392c99567b/apps/site/src/app/events/getEvents.ts#L12-L25) as an example. There are also existing types for Sanity objects, such as the image, that you can use (also used in the example).

Here's an article covering [the advantages of this approach](https://www.simeongriggs.dev/type-safe-groq-queries-for-sanity-data-with-zod).

Reply:
After parsing with zod, the object will also be giving a typescript type. This will make the `// @ts-expect-error` in [page.tsx on line 51](https://github.com/HackAtUCI/hack.ics.uci.edu/blob/4802377eb66c76732584d7708f356f392c99567b/apps/site/src/app/about/page.tsx#L51C8-L51C27) unnecessary.